### PR TITLE
Turn off camera light

### PR DESCRIPTION
**Pull request by golgetahir**
This PR fixes #https://github.com/ant-media/Ant-Media-Server/issues/3420

### CODE REVIEW

File: src/main/webapp/js/webrtc_adaptor.js [hunk 1]
```diff
@@ -1289,31 +1294,35 @@ export class WebRTCAdaptor
...
-			}, false);
-		}
-		else if (this.remotePeerConnection != null) {
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = true;
-		}
-	}
+	turnOffLocalCamera(streamId) 
+	 {
+		 this.dummyCanvas.getContext('2d').fillRect(0, 0, 320, 240);
```
Comment (mekya) [inline src/main/webapp/js/webrtc_adaptor.js:1299]: I think it should be compatible with older version. Shall we think a way of doing that?

**Review by mekya — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/webrtc_adaptor.js [hunk 2]
```diff
@@ -1292,31 +1299,53 @@ export class WebRTCAdaptor
...
+			this.dummyCanvas.getContext('2d').fillRect(0, 0, 320, 240);
+		 	this.replacementStream = this.dummyCanvas.captureStream();
+		}, 3000);
+
+		 if (this.remotePeerConnection != null) {
+			 if(streamId != null || streamId != undefined){
+				this.updateVideoTrack(this.replacementStream, streamId, this.mediaConstraints, null, true);
+			 }
+			 else{
+				this.updateVideoTrack(this.replacementStream, this.publishStreamId, this.mediaConstraints, null, true);
```
Comment (mekya) [inline src/main/webapp/js/webrtc_adaptor.js:1316]: Make a single call to `updateVideoTrack` by the correct parameter. Calling the same method in two different places for the same purpose is not a good practice

File: src/main/webapp/js/webrtc_adaptor.js [hunk 3]
```diff
@@ -1292,31 +1299,53 @@ export class WebRTCAdaptor
...
+		}
+		 if(this.localStream == null){
+			 this.navigatorUserMedia(this.mediaConstraints, stream =>{
+				 this.gotStream(stream);
+			 }, false);
+		 }
+		 //This method will get the camera track and replace it with dummy track
+		 else if (this.remotePeerConnection != null) {
+			 this.navigatorUserMedia(this.mediaConstraints, stream =>{
+				if(streamId != null || streamId != undefined){
```
Comment (mekya) [inline src/main/webapp/js/webrtc_adaptor.js:1337]: Same comments for turnOnLocalCamera

**Review by mekya — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/webrtc_adaptor.js [hunk 4]
```diff
@@ -1290,33 +1297,65 @@ export class WebRTCAdaptor
...
-		}
-		else if (this.remotePeerConnection != null) {
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = true;
-		}
-	}
+	turnOffLocalCamera(streamId) 
+	 {
+		 //Initialize the first dummy frame for switching.
+		this.initializeDummyFrame();
```
Comment (mekya) [inline src/main/webapp/js/webrtc_adaptor.js:1308]: If user accidently calls the `turnOffLocalCamera` several times, it creates several canvas and timers

**Review by mekya — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/webrtc_adaptor.js (hunk 4 above)
Comment (golgetahir) [reply src/main/webapp/js/webrtc_adaptor.js:1308]: Canvas is created on the constructor only once and if a new frame is created previous one is released from the same variable. About timers I have refactored it.

**Review by mekya — APPROVED** [review]
I've added another commit

**Pull request merged**